util/util.go

```diff
@@ -135,13 +135,18 @@ func MustParseAddr(str string) ma.Multiaddr {
 	return addr
 }
 
-func SchemaFromInstance(i interface{}) string {
-	schema := jsonschema.Reflect(i)
-	JSON, err := json.MarshalIndent(schema, "", "  ")
-	if err != nil {
+func SchemaFromInstance(i interface{}, expandedStruct bool) *jsonschema.Schema {
```

Comment:
Most of the noise in this PR is because of adding the `expandedStruct` option here. When set to `true`, the generated schema will use top level type definitions instead of using a ref to a nested definition. We need to be able to test both cases.